backend/core/views.py

```diff
@@ -300,29 +298,33 @@ def competence_reviews_outstanding(self, request, pk=None):
 #     return Response(result)
 
 
-@staff_member_required
-def bulk_add_users_to_team(request, team_id):
-    team = get_object_or_404(Team, pk=team_id)
-
-    if request.method == "POST":
-        form = BulkAddUsersForm(request.POST)
-        if form.is_valid():
-            email_addresses = form.cleaned_data["email_addresses"]
-            users_added_to_team = add_users_to_team(team, email_addresses)
-            if users_added_to_team:
-                messages.success(
-                    request,
-                    f"The following users were successfully added to the \"{team.name}\" team: {', '.join(users_added_to_team)}",
-                )
-                return redirect(f"/admin/core/team/{team_id}/change")
-            else:
-                messages.error(
-                    request,
-                    f'No users were added to the "{team.name}" team. Make sure the users and or entered email addresses exist and try again.',
-                )
-    else:
-        form = BulkAddUsersForm()
-
-    return render(
-        request, "admin/core/bulk_add_users_form.html", {"form": form, "team": team}
-    )
+class BulkAddUsersToTeamView(LoginRequiredMixin, FormView):
+    form_class = BulkAddUsersToTeamForm
+    template_name = "admin/core/bulk_add_users_form.html"
+    success_url = reverse_lazy("bulk_add_users_to_team")
+
+    def get_login_url(self):
+        return reverse("admin:login")
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        self.team = Team.objects.get(id=self.kwargs["team_id"])
```

Comment:
It'd be nicer if we handled a scenario where id is non existent in the db because the `.get()` method, unlike `filter()` will raise an exception if the team ID does not exist.